aws_topology/stackstate_checks/aws_topology/resources/route53_hostedzone.py

```diff
@@ -1,40 +1,104 @@
-from .utils import make_valid_data
+from .utils import (
+    make_valid_data,
+    create_arn as arn,
+    client_array_operation,
+    set_required_access_v2,
+    transformation,
+)
 from .registry import RegisteredResourceCollector
+from collections import namedtuple
+from schematics import Model
+from schematics.types import StringType, ModelType, ListType
+
+
+def create_arn(region=None, account_id=None, resource_id=None, **kwargs):
+    return arn(resource="route53", region="", account_id="", resource_id="hostedzone/" + resource_id)
+
+
+class HostedZoneInfo(Model):
+    Id = StringType(required=True)
+    Name = StringType(default="UNKNOWN")
+
+
+class DelegationSet(Model):
+    Id = StringType()
+
+
+class VPCs(Model):
+    VPCId = StringType()
+
+
+class HostedZone(Model):
+    HostedZone = ModelType(HostedZoneInfo, required=True)
+    DelegationSet = ModelType(DelegationSet)
+    VPCs = ListType(ModelType(VPCs))
+
+
+HostedZoneData = namedtuple("HostedZoneData", ["hosted_zone", "tags", "resource_record_sets"])
 
 
 class Route53HostedzoneCollector(RegisteredResourceCollector):
     API = "route53"
     API_TYPE = "global"
     COMPONENT_TYPE = "aws.route53.hostedzone"
 
+    @set_required_access_v2("route53:ListTagsForResource")
+    def collect_tags(self, hosted_zone_id):
+        return (
+            self.client.list_tags_for_resource(ResourceType="hostedzone", ResourceId=hosted_zone_id)
```

Comment:
Tags collections often do support paginate. We need to check this for all APIs. Often they have a "HasMoreTags" attribute. Don't know if that supports the can_paginate call.

Reply:
This particular resource [does not](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/route53.html#Route53.Client.list_tags_for_resource) - let's worry about this with a pass over all resources later